security/pkg/pki/signingapi/client.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signingapi
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/api/security/v1alpha1"
+	"istio.io/istio/pkg/mcp/creds"
+	"istio.io/pkg/log"
+)
+
+var (
+	signingLog = log.RegisterScope("SingingAPIClient", "", 0)
+)
+
+const (
+	defaultRequestTimeout = 10
+)
+
+// Client create Istio Certificate Signing Client
+type Client struct {
+	Address          string
+	RequestTimeout   time.Duration
+	Conn             *grpc.ClientConn
+	ClientV1         v1alpha1.IstioCertificateServiceClient
+	TLSClientSetting *networking.ClientTLSSettings
+}
+
+// New create new Istio Signing Certificate Client
+func New(address string, tlsSetting *networking.ClientTLSSettings, requestTimeout time.Duration) *Client {
+	// default timeout is 10s
+	if requestTimeout.Seconds() < 1 {
```

Comment:
why not == 0? couldn't the timeout be 500ms

Reply:
good catch!
Thanks @howardjohn